buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/TestClustersPlugin.java

```diff
@@ -98,7 +116,15 @@ public void apply(Project project) {
             // the clusters will look for artifacts there based on the naming conventions.
             // Tasks that use a cluster will add this as a dependency automatically so it's guaranteed to run early in
             // the build.
-            rootProject.getTasks().create(SYNC_ARTIFACTS_TASK_NAME, SyncTestClustersConfiguration.class);
+            rootProject.getTasks().create(SYNC_ARTIFACTS_TASK_NAME, Sync.class, sync -> {
+                sync.from((Callable<List<FileTree>>) () ->
```

Comment:
wondering why the cast is needed here

Reply:
`CopySpec.from()` simply takes `Object...` as an argument. Each item passed can be any of the types listed [here](https://docs.gradle.org/current/javadoc/org/gradle/api/Project.html#files-java.lang.Object...-). We can't just pass a use raw lambda, as Java has now idea what SAM type to cast it to. So we have to explicitly cast it as one of the supported types. We don't have this problem in Groovy as `{ }` returns an instance of `Closure`.

There's actually no real need to include the generic type arguments here, Gradle will figure this out at runtime, but its more explicit and folks reading the code immediately see what that lambda returns.

Reply:
Additionally, if it's not obvious, we wrap all this in a `Callable` because we are iterating over a `Configuration` here so we can return a `FileTree` from each resolved artifact. Doing so causes the Configuration to be resolved (and files to be downloaded) so we want to defer this until task execution. Or in this case more specifically, during task graph calculation.